deploy.bat

```diff
@@ -0,0 +1,107 @@
+@rem This script should be called with both the current version number of the
+@rem server. It should generally not be called manually, but rather by using the
+@rem NPM scripts (`yarn deploy` or `yarn deploystaging`).
+@rem
+@rem Since it is intended for building deployable artifacts, it requires a clean
+@rem git working directory (e.g. no uncommitted changes).
+@rem
+@rem Example of running it manually:
+@rem  - "deploy.bat 6.1.2"
+@rem  - "deploy.bat $npm_package_version" (if run from a npm script)
+@echo off
+
+SETLOCAL
+
+if "%1"=="" (
+  echo Missing required version argument
+  goto :exit
+)
+
+SET version=%1
+
+@rem Handle changes that are not staged
+git diff --exit-code > nul
+if errorlevel 1 (
+  echo Uncommitted git changes detected. Please commit them and try again.
+  goto exit
+)
+
+@rem Handle changes that are staged
+git diff --cached --exit-code > nul
+if errorlevel 1 (
+  echo Uncommitted git changes detected. Please commit them and try again.
+  goto exit
+)
+
+
+SET gitsha=
+FOR /F "tokens=* USEBACKQ" %%F IN (`git rev-parse --verify HEAD`) DO (
+  SET gitsha=%%F
+)
+
+if errorlevel 1 (
+  echo Error retrieving current git SHA
+  goto :exit
+)
+if "%gitsha%" == "" (
+  echo Error retrieving current git SHA
+  goto :exit
+)
+
+SET NAME=shieldbattery/shieldbattery
+SET IMG=%NAME%:%gitsha%
+SET NPM_TAG=%NAME%:%version%
+SET LATEST_TAG=%NAME%:latest
+
+@rem The docker file will take whatever files are currently built and create a docker image out of
+@rem them. This means that this script doesn't care which branch you're trying to deploy. It only
+@rem cares whether you're trying to deploy to the main production server or the staging server.
```

Comment:
This sentence can also be removed I believe.

Reply:
I think it's still relevant, it is indeed branch-agnostic

Reply:
I meant the last sentence of this comment, i.e.
```
It only cares whether you're trying to deploy to the main production server or the staging server.
```

I don't think this script cares about that anymore :d

Reply:
Ah, I was thinking about a different comment elsewhere anyway, I don't think this one is particularly useful so I'll just remove it.